src/main/java/br/ufpe/cin/mergers/SemistructuredMerge.java

```diff
@@ -107,167 +119,228 @@ public static String merge(File left, File base, File right, MergeContext contex
+			SuperimpositionStep step) {
+		if (!nodeA.compatibleWith(nodeB))
+			return null;
 
-				/*
-				 * nodes from left or leftBase
-				 */
-				List<FSTNode> nonterminalAChildren = nonterminalA.getChildren();
-				
-				for (int i = 0; i < nonterminalAChildren.size(); i++) {
-					FSTNode childA = nonterminalAChildren.get(i);
-					FSTNode childB = nonterminalB.getCompatibleChild(childA);
-					
-					if (childB == null) { 								// is a new node from left, or a deleted base node in right
-						FSTNode cloneA = childA.getDeepClone();
-						if (childA.index == -1)
-							childA.index = nodeA.index;
-						cloneA.index = childA.index;
-
-						FSTNode childALeftNeighbour = getLeftNeighbourNode(nonterminalAChildren, i);
-						FSTNode childARightNeighbour = getRightNeighbourNode(nonterminalAChildren, i);
-						addNodeToNonTerminalNearNeighbour(cloneA, childALeftNeighbour, childARightNeighbour, nonterminalComposed);
-
-						if (context.deletedBaseNodes.contains(childA)) { // this is only possible when processing right nodes because this is a base node not present either in left and right
-							context.deletedBaseNodes.remove(childA);
-							context.deletedBaseNodes.add(cloneA);
-						}
-
-						if(isProcessingBaseTree){ //node added by left in relation to base
-							context.addedLeftNodes.add(cloneA);
-						} else {
-							if(!context.addedLeftNodes.contains(childA))
-								context.nodesDeletedByRight.add(cloneA);
-						}
-					} else {
-						if (!isProcessingBaseTree) {
-							context.deletedBaseNodes.remove(childA); 	// node common to right and base but not to left
-						}
-					}
-				}
-				return nonterminalComposed;
+		// Setting up superimposed node.
+		FSTNode result = nodeA.getShallowClone();
+		result.index = nodeB.index;
+		result.setParent(parent);
+
+		if (areBothTerminals(nodeA, nodeB, parent)) {
+			FSTTerminal terminalA = (FSTTerminal) nodeA;
+			FSTTerminal terminalB = (FSTTerminal) nodeB;
+			FSTTerminal terminalResult = (FSTTerminal) result;
+
+			return superimposeTerminals(terminalA, terminalB, step, terminalResult);
+		}
+
+		else if (areBothNonTerminals(nodeA, nodeB)) {
+			FSTNonTerminal nonTerminalA = (FSTNonTerminal) nodeA;
+			FSTNonTerminal nonTerminalB = (FSTNonTerminal) nodeB;
+			FSTNonTerminal nonTerminalResult = (FSTNonTerminal) result;
+
+			return superimposeNonTerminals(nonTerminalA, nonTerminalB, context, step, nonTerminalResult);
+		}
+
+		return null;
+	}
+
+	private static boolean areBothTerminals(FSTNode nodeA, FSTNode nodeB, FSTNonTerminal parent) {
+		return nodeA instanceof FSTTerminal && nodeB instanceof FSTTerminal;
+	}
+
+	private static boolean areBothNonTerminals(FSTNode nodeA, FSTNode nodeB) {
+		return nodeA instanceof FSTNonTerminal && nodeB instanceof FSTNonTerminal;
+	}
+
+	private static FSTNode superimposeTerminals(FSTTerminal terminalA, FSTTerminal terminalB, SuperimpositionStep step,
+			FSTTerminal result) {
+		if (!terminalA.getMergingMechanism().equals("Default")) {
```

Comment:
@jvcoutinho @guilhermejccavalcanti quando não é default, não deveria tomar outra ação aqui mesmo? essa ação é feita em outro lugar? algum handler?

Reply:
Não é feita nenhuma ação. Esse código provavelmente é o original do FST, que lida com esse MergeMechanism na gramática. Eu imagino que essa condição possa ser retirada.

Reply:
Na prática, o default prevalece a contribuição de left. É herança do FST. Acredito que todos os nós anotados como "Default" poderiam ser anotados como "Linebased" para invocar o textual, mas assim, haveria lugares em que o resultado é trivial (três nós com o mesmo conteúdo), chamar o textual num caso desse é recurso desnecessário. 

A gramática como tá hoje é estável, mais no início observamos problemas causados por esses "default" mas foram sendo corrigidos com o passar do tempo, até a versão atual.